pkg/flow/internal/controller/loader.go

```diff
@@ -345,6 +345,17 @@ func (l *Loader) populateServiceNodes(g *dag.Graph, serviceBlocks []*ast.BlockSt
 	// Now, assign blocks to services.
 	for _, block := range serviceBlocks {
 		blockID := BlockComponentID(block).String()
+
+		if l.isModule() {
```

Comment:
Will this allow some components to still be running even with diags and what the impact is?

Reply:
On init, if the module contains a service block the agent will just exit and log this error.
If the agent is running and the module content is updated, the reload of the module will fail. As a result, the module will be marked as unhealthy and the content of the last successful load will be used. There should be no impact on the running components.